src/main/java/org/candlepin/util/OIDUtil.java

```diff
@@ -28,15 +29,20 @@ public final class OIDUtil {
     private OIDUtil() {
     }
 
-    public static final String REDHAT_OID = "1.3.6.1.4.1.2312.9";
-    public static final String SYSTEM_NAMESPACE_KEY = "System";
-    public static final String ORDER_NAMESPACE_KEY = "Order";
-    public static final String CHANNEL_FAMILY_NAMESPACE_KEY = "Channel Family";
-    public static final String ROLE_ENT_NAMESPACE_KEY = "Role Entitlement";
-    public static final String PRODUCT_CERT_NAMESPACE_KEY = "Product";
-    public static final String ENTITLEMENT_VERSION_KEY = "Entitlement Version";
-    public static final String ENTITLEMENT_DATA_KEY = "Entitlement Data";
-    public static final String ENTITLEMENT_TYPE_KEY = "Entitlement Type";
+    private static final String REDHAT_OID = "1.3.6.1.4.1.2312.9";
+
+    public enum Namespace {
```

Comment:
This is interesting. On the surface I like this, because it gets rid of the magic strings and makes the code far safer everywhere down the line.

However, once we have an enum, we're at the point where we may not even *need* the map to do the OID conversion. It could be a case where we have a generic enum interface for like `OIDComponent` which defines a `toOidString()` method that all of our enums could implement. Then `OIDUtil` can get paired down to a single method along the lines of `build(OIDComponent...)` which would end up being basically just a join of all the components in the order specified.

Even better is if it built to an OID itself and that could be passed to our extensions. But this is getting into the crypto layer refactor which may or may not even be necessary in the near term.

Reply:
I agree. There's quite a lot we could do to improve OID mapping. Separate card is a good idea. I didn't want to push this refactoring too far as it would drown the actual change that's pretty small.